storage/Filesystems/BtrfsImpl.cc

```diff
@@ -505,18 +508,159 @@ namespace storage
     }
 
 
-#if 0
     ResizeInfo
-    Btrfs::Impl::detect_resize_info_on_disk() const
+    Btrfs::Impl::detect_resize_info(const BlkDevice* blk_device) const
+    {
+	ResizeInfo default_resize_info(true, 0, min_size(), max_size());
+
+	if (!exists_in_system())
+	{
+	    // The filesystem does not exist on disk yet.
+
+	    if (get_blk_devices().size() == 1)
+	    {
+		// Rely on BlkFilesystem::Impl when the filesystem is a not committed
+		// single-device Btrfs.
+	
+		return BlkFilesystem::Impl::detect_resize_info();
+	    }
+	    else
+	    {
+		// The filesystem is a not committed multi-device Btrfs.
+
+		return default_resize_info;
+	    }
+	}
+	else
+	{
+	    // The filesystem already exists on disk.
+
+	    const Btrfs* filesystem = redirect_to_system(get_non_impl());
+
+	    if (filesystem->get_blk_devices().size() == 1)
+	    {
+		// Currently, the filesystem is a single-device Btrfs.
+		
+		if (blk_device && blk_device->get_sid() != filesystem->get_impl().get_blk_device()->get_sid())
```

Comment:
Hm... can we be sure that the blk_device that was passed actually belongs to this device?
I trust nobody, in particular not anybody calling my function... :smile:

Reply:
Well, for now, this parameter only contains a parent of the filesystem. But yes, I would add an extra check.